### PR TITLE
Add namespaceDefinitionType to standardSync

### LINKED ISSUE #3481: Ability to override destination namespace.
## Tell us about the problem you're trying to solve
Today we mirror the source's namespace when syncing data.

This dedups on conflicting table names e.g. a `users` table in the `public` schema and a `users` table in a `staging` schema is replicated into `public` and `staging` schema respectively.

However, this is too inflexible in some cases where users want more logical naming over the replicated tables instead of the source's schema.

## Describe the solution you’d like
User should be able to configure the destination namespace at a table level when setting up a connection.

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1200367912513076/1200368196361379) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by ChristopheDuong**
## What
Implements backend part of #3481 

## How
Expose an option on how to use the source namespaces in the destination:
1. Ignore it: (Always use default namespace from destination settings)
2. Use it: (Always use namespace from source when available, fallback to default destination namespace if unavailable)

## Pre-merge Checklist
- [x] *Run integration tests*
- [ ] *Publish Docker images*

## Recommended reading order
1. `airbyte-workers/src/main/java/io/airbyte/workers/protocols/airbyte/NamespacingMapper.java`
1. `airbyte-api/src/main/openapi/config.yaml`
1. the rest

see https://docs.google.com/document/d/1qFk4YqnwxE4MCGeJ9M2scGOYej6JnDy9A0zbICP_zjI/edit# for more context

### CODE REVIEW

File: airbyte-migration/src/main/java/io/airbyte/migrate/migrations/MigrationV0_25_0.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Consumer;
+import java.util.stream.Stream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+// This migration does the following:
+// 1. Add fields to StandardSync.
+public class MigrationV0_25_0 extends BaseMigration implements Migration {
```
Comment (davinchia) [inline airbyte-migration/src/main/java/io/airbyte/migrate/migrations/MigrationV0_25_0.java:42]: you preempted my migration question!

File: airbyte-api/src/main/openapi/config.yaml [hunk 2]
```diff
@@ -2096,6 +2102,13 @@ components:
             - days
             - weeks
             - months
+    NamespaceDefinitionType:
```
Comment (davinchia) [inline airbyte-api/src/main/openapi/config.yaml:2105]: discussed offline: we should leave a comment here explaining the default behaviour

File: airbyte-workers/src/main/java/io/airbyte/workers/protocols/airbyte/NamespacingMapper.java [hunk 3]
```diff
@@ -40,30 +41,38 @@
...
-    catalog.getStreams().forEach(s -> mutateStream(s.getStream(), prefix));
+    catalog.getStreams().forEach(s -> mutateStream(s.getStream(), namespaceDefinition, prefix));
     return catalog;
   }
 
   @Override
   public AirbyteMessage mapMessage(final AirbyteMessage inputMessage) {
     if (inputMessage.getType() == Type.RECORD) {
       final AirbyteMessage message = Jsons.clone(inputMessage);
+      if (namespaceDefinition != null && namespaceDefinition.equals(NamespaceDefinitionType.DESTINATION)) {
```
Comment (davinchia) [inline airbyte-workers/src/main/java/io/airbyte/workers/protocols/airbyte/NamespacingMapper.java:63]: spoke offline: same comment here about default behaviour possibly linking to the api spec file.

**Review by davinchia — APPROVED** [review]
Looks great!

One comment about better explanation around the default behaviour. Feel free to merge whenever.

**Review by cgardens — APPROVED** [review]
looks good!

**Pull request merged**